### DISCUSSION TITLE
SMTP Settings

### DISCUSSION THREAD

**Discussion by HidYn** [mediacms-io/mediacms#194] [Q&A]
Hi,

Thanks for such a fantastic application. I'm trying to setup SMTP to an e-mail server that accepts anonymous connections (Windows based) which requires e-mails to be sent from a certain sender address.

DEFAULT_FROM_EMAIL = 'senderaddress@test.com'
EMAIL_HOST_PASSWORD = ''
EMAIL_HOST_USER = ''
EMAIL_USE_TLS = False
SERVER_EMAIL = DEFAULT_FROM_EMAIL
EMAIL_HOST = 'smtpserver.domain.com'
EMAIL_PORT = 25
ADMIN_EMAIL_LIST = ['info@mediacms.io']

I have set the above setting and keep getting connection refused.

ConnectionRefusedError: [Errno 111] Connection refused

Any help would be appreciated.

**Comment by Dan1ell** [comment]
Did you try it yet with some Gmail credentials and see if it connects to that?

DEFAULT_FROM_EMAIL = (your Gmail address)
EMAIL_HOST_PASSWORD = (your Gmail password)
EMAIL_HOST_USER = (your Gmail username)
EMAIL_USE_TLS: True
SERVER_EMAIL = DEFAULT_FROM_EMAIL
EMAIL_HOST = smtp.gmail.com
EMAIL_PORT = 587
ADMIN_EMAIL_LIST = ['(your Gmail address)']

Rerstart the webserver.

**Reply by HidYn** [reply]
Thank you for the reply.

It's difficult to test Gmail SMTP sadly as I work in a hospital and the only way we can send e-mail is via our internal SMTP server which then forwards through a service called NHSMail. 

Complicated sadly.

**Comment by mgogoulos** [comment]
hi, 
a) make sure that email settings are correct, and 
b) outgoing traffic is not blocked by some firewall

Are you using the docker setup or the single server one? For the single server you have to put these setting on cms/local_settings.py, while on the dockerized setup on file deploy/docker/local_settings.py

Make sure settings work, you may also try to login to the shell and make a test
For example for the single server installation, open the shell 

```
source  /home/mediacms.io/bin/activate
python manage.py shell
```

```
from django.conf import settings 
from django.core.mail import EmailMessage 
email = EmailMessage( 
    'title', 
    'test message', 
    settings.DEFAULT_FROM_EMAIL, 
    settings.ADMIN_EMAIL_LIST, 
    reply_to=[settings.DEFAULT_FROM_EMAIL], 
) 
email.send(fail_silently=False)
Out[1]: 1 # 1 means email was sent!
```

Or if you're running the docker setup, bash into a container and do the same

```
root@server:~# docker ps
CONTAINER ID   IMAGE                       COMMAND                  CREATED      STATUS                PORTS                                                                      NAMES
70874a57e7fc   mediacms/mediacms:latest    "./deploy/docker/ent…"   5 days ago   Up 5 days             80/tcp, 9000/tcp                                                           mediacms_celery_worker_1

root@server:~# docker exec -it mediacms_web_1 bash
root@6576df0fab12:/home/mediacms.io/mediacms# python manage.py shell

from django.conf import settings 
from django.core.mail import EmailMessage 
email = EmailMessage( 
    'title', 
    'test message', 
    settings.DEFAULT_FROM_EMAIL, 
    settings.ADMIN_EMAIL_LIST, 
    reply_to=[settings.DEFAULT_FROM_EMAIL], 
) 
email.send(fail_silently=False)

```

**Reply by HidYn** [reply]
Thank you for the reply, really appreciate the help.

I've set it up using the single server method using Ubuntu 20.04 as per install instructions.

There is no firewall - Running it in a vmware enviroment and trying to send to a windows server that accepts anonymous connections as long as the IP is whitelisted which it is.

I am editing cms/local_settings.py.

![image](https://user-images.githubusercontent.com/31063223/119315981-52697700-bc6e-11eb-8ce5-475cdef26bdf.png)

**Reply by devilmind81** [reply]
Hi @mgogoulos,
After I have executed the steps on below I'm obtaining a different output and the email was not sent, what can i check further?

```
root@localhost:/home/mediacms.io/mediacms/cms# docker exec -it  mediacms_web_1 /bin/bash
root@bdd978e6afbb:/home/mediacms.io/mediacms# source  /home/mediacms.io/bin/activate
(mediacms.io) root@bdd978e6afbb:/home/mediacms.io/mediacms# python manage.py shell
Python 3.8.14 (default, Sep 13 2022, 02:23:58)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from django.conf import settings
>>> from django.core.mail import EmailMessage
>>> email = EmailMessage(

    reply_to=[settings.DEFAULT_FROM_EMAIL],
)
...     'title',
...     'test message',
...     settings.DEFAULT_FROM_EMAIL,
...     settings.ADMIN_EMAIL_LIST,
email.send(fail...     reply_to=[settings.DEFAULT_FROM_EMAIL],
... )
>>> email.send(fail_silently=False)
[<AsyncResult: 6f845e5e-7029-4a24-8ae2-3af9d67fdb7e>]
```

**Reply by grubinstein** [reply]
I have the same problem as you @devilmind81. Did you find a solution?

**Reply by devilmind81** [reply]
Unfortunately I cannot found a solution for it yet....I missing the possibility to permit all user to auto-register for the moment.

**Reply by bananacy** [reply]
I think i have same problem. unable to send emails : 
`
[2023-05-24 00:06:31,364: ERROR/ForkPoolWorker-7] djcelery_email_send_multiple[068d740a-754d-4a2f-854f-3403173fcebf]: Cannot reach CELERY_EMAIL_BACKEND django.core.mail.backends.smtp.EmailBackend
Traceback (most recent call last):
  File "/home/mediacms.io/lib/python3.8/site-packages/djcelery_email/tasks.py", line 40, in send_emails
    conn.open()
  File "/home/mediacms.io/lib/python3.8/site-packages/django/core/mail/backends/smtp.py", line 62, in open
    self.connection = self.connection_class(self.host, self.port, **connection_params)
  File "/usr/lib/python3.8/smtplib.py", line 255, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python3.8/smtplib.py", line 341, in connect
    (code, msg) = self.getreply()
  File "/usr/lib/python3.8/smtplib.py", line 398, in getreply
    raise SMTPServerDisconnected("Connection unexpectedly closed")
smtplib.SMTPServerDisconnected: Connection unexpectedly closed
`

**Reply by Randymations** [reply]
Specifically for the "Connection unexpectedly closed" issue, for anyone finding this in the future, if you're using TLS/SMTPS, in your `local_settings.py`, try setting
```py
EMAIL_USE_TLS = False
EMAIL_USE_SSL = True
```
This worked for me.

**Comment by gam-ragnar** [comment]
Getting a "Cannot reach CELERY_EMAIL_BACKEND django.core.mail.backends.smtp.EmailBackend" my new  setup is @swiftugandan [letsencrypt docker-compose ](https://github.com/mediacms-io/mediacms/pull/178). Error log posted below:
```
[2021-06-01 15:36:15,181: ERROR/ForkPoolWorker-7] djcelery_email_send_multiple[86d2d4ce-5b56-45a5-b062-a3f22cd4a29b]: Cannot reach CELERY_EMAIL_BACKEND django.core.mail.backends.smtp.EmailBackend
Traceback (most recent call last):
  File "/home/mediacms.io/lib/python3.8/site-packages/djcelery_email/tasks.py", line 40, in send_emails
    conn.open()
  File "/home/mediacms.io/lib/python3.8/site-packages/django/core/mail/backends/smtp.py", line 62, in open
    self.connection = self.connection_class(self.host, self.port, **connection_params)
  File "/usr/local/lib/python3.8/smtplib.py", line 255, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/local/lib/python3.8/smtplib.py", line 339, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/local/lib/python3.8/smtplib.py", line 310, in _get_socket
    return socket.create_connection((host, port), timeout,
  File "/usr/local/lib/python3.8/socket.py", line 808, in create_connection
    raise err
  File "/usr/local/lib/python3.8/socket.py", line 796, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address
[2021-06-01 15:36:15,187: WARNING/ForkPoolWorker-7] djcelery_email_send_multiple[86d2d4ce-5b56-45a5-b062-a3f22cd4a29b]: Failed to send email message to ['***@gmail.com'], retrying. (OSError(99, 'Cannot assign requested address'))
```

**Reply by gam-ragnar** [reply]
Hi @mgogoulos,  should I create an issue ticket for above error?
